### ISSUE TITLE
[BUG] Firefox multi page screenshots and focus ring

### ISSUE THREAD

**Issue by benoitjchevalier** [microsoft/playwright#3152]
**Context:**
- Playwright Version: 1.2.1
- Operating System: Mac
- Node version: 12.14.1
- Browser: Firefox, Webkit

**Code Snippet**
https://github.com/benoitjchevalier/playwright-focus

npm install
node run.js

Observe lack of focus ring on some FF screenshots

**Describe the bug**

Hello there 👋 . We're building an in-house visual regression tool using playwright which has been a huge help for us. I've recently come across an edge case on Firefox and sometimes Webkit.

When running multiple pages (each with a unique context) of the same browser in parallel and taking screenshots of a focused element then on FF the screenshot will miss the focus ring.

Chrome always behaves and I've seen rare occurrences where webkit would also suffer from the focusringlessness.
The element itself is correctly identified as active by the browser when testing through `element === (document|shadowRoot).activeElement`, before and after the screenshot.

I've created a small repro in the above repo. Please note that this repo does everything from the node context where our tool runs the tests in the browser context and the notifies the node context when a screenshot should be taken. So the difference is that we focus from the browser context where the reproduction focuses from the playwright elementHandle. That said the results are similar.

I'm hoping this can be fixed at the playwright level and isn't a limitation at the browser level :)

**Comment by aslushnikov** [comment]
@benoitjchevalier Hmm this does look like a dupe of https://github.com/microsoft/playwright/issues/3476

**Comment by benoitjchevalier** [comment]
Given the timeline isn't the other one a duplicate of this one? ;)
good to know it's being looked at though

**Comment by dgozman** [comment]
Closing as duplicate of #3476.
